api/scpca_portal/test/models/test_job.py

```diff
@@ -6,62 +6,96 @@
         self.assertEqual(Job.objects.count(), 1)
         saved_job = Job.objects.first()
-        self.assertEqual(saved_job.batch_job_name, self.batch_job_name)
+        self.assertEqual(saved_job.batch_job_id, self.mock_batch_job_id)
         self.assertEqual(saved_job.batch_job_queue, settings.AWS_BATCH_JOB_QUEUE_NAME)
         self.assertEqual(saved_job.batch_job_definition, settings.AWS_BATCH_JOB_DEFINITION_NAME)
-        self.assertEqual(
-            saved_job.batch_container_overrides, self.project_job_batch_container_overrides
-        )
-        self.assertIn(self.project_id, saved_job.batch_container_overrides["command"])
+        self.assertIn("--project-id", saved_job.batch_container_overrides["command"])
+        self.assertIn(self.mock_project_id, saved_job.batch_container_overrides["command"])
         self.assertEqual(saved_job.state, JobStates.SUBMITTED)
         self.assertIsInstance(saved_job.submitted_at, datetime)
+
+    @patch("scpca_portal.models.Job._batch")
+    def test_terminate_job(self, mock_batch_client):
+        # Set up mock for the SUBMITTED job
+        submitted_job = JobFactory(
+            batch_job_name=self.mock_project_batch_job_name,
+            batch_job_id=self.mock_batch_job_id,
+            state=JobStates.SUBMITTED,
+        )
+
+        success = submitted_job.terminate(retry_on_termination=True)
+        mock_batch_client.terminate_job.assert_called()
+        self.assertTrue(success)
+
+        # After termination, the job should be saved with correct field values
+        saved_job = Job.objects.first()
+        self.assertEqual(saved_job.state, JobStates.TERMINATED)
+        self.assertTrue(saved_job.retry_on_termination)
+        self.assertIsInstance(saved_job.terminated_at, datetime)
+
+        # Set up mock for the TERMINATED job
+        terminated_job = JobFactory(
+            batch_job_name=self.mock_project_batch_job_name,
+            batch_job_id=self.mock_batch_job_id,
+            state=JobStates.TERMINATED,
+        )
+
+        # Should return Ture for previously terminated job
+        success = terminated_job.terminate(retry_on_termination=True)
+        mock_batch_client.terminate_job.assert_called()
+        self.assertTrue(success)
+
+    @patch("scpca_portal.models.Job._batch")
+    def test_terminate_job_failure(self, mock_batch_client):
+        job = JobFactory(
+            batch_job_name=self.mock_project_batch_job_name,
+            batch_job_id=self.mock_batch_job_id,
+            state=JobStates.SUBMITTED,
+        )
+
+        # Set up mock to raise an exception
+        mock_batch_client.terminate_job.side_effect = Exception("Exception")
+
+        success = job.terminate(retry_on_termination=True)
+        mock_batch_client.terminate_job.assert_called()
+        self.assertFalse(success)
+
+        # The job should not be updated
+        saved_job = Job.objects.first()
+        self.assertEqual(saved_job.state, JobStates.SUBMITTED)
```

Comment:
```suggestion
        self.assertEqual(saved_job.state, job.state)
```

Let's assert that the value hasn't changed by checking against the un-refreshed value